### DISCUSSION TITLE
bridged networking deprecation: For the love of God, Montresor!

### DISCUSSION THREAD

**Discussion by cerebrate** [microsoft/WSL#12345] [General] [answered]
From [the 2.4.5 release notes](https://github.com/microsoft/WSL/releases/tag/2.4.5):

> Add user warning letting users know bridged networking will be deprecated

Why is this being deprecated?

And more importantly, can we please not?

(I have tried the mirrored networking scheme, and all I have achieved by doing thus far is make an unholy and non-working mess of my WSL network configuration in an attempt to retain the functionality I have now. 'Tis enough to drive a man to the consumption of an entire cask of Amontillado.)

**Comment by OneBlue** [comment, marked as answer]
@cerebrate: Just to give closure here, we did bring bridged networking back in [2.5.6](https://github.com/microsoft/WSL/releases/tag/2.5.6) 

The context on the deprecation is that the original implementation of bridged networking would have made [open sourcing](https://blogs.windows.com/windowsdeveloper/2025/05/19/the-windows-subsystem-for-linux-is-now-open-source/) difficult, but given the user feedback we investigated more and found a different way to implement bridged networking, which allowed us to both open source and keep bridged networking !

**Reply by cerebrate** [reply]
Most excellent, indeed.

(As I said over in #12743, there are _so many_ things enabled by bridged mode that make WSL an awesome sysadmin platform, so I'm very glad we get to keep all that and get open source too!)
